### ISSUE TITLE
out of date fluidsynth library?

### ISSUE THREAD

**Issue by DaforLynx** [Birch-san/juicysfplugin#21]
![image](https://user-images.githubusercontent.com/13303368/99127963-6d23aa80-25be-11eb-9f65-1933c35dc733.png)
I just started using Ardour 6.3 in order to get program changes imported from MIDI files to work. However, when I scan for VSTs, Ardour throws this error when it gets to Juicy SF. I suspect it might be because something was either added or taken out of libfluidsynth-2.dll recently that the Ardour folks used. Or maybe not, I am no programmer. I see that you have been active relatively recently and I hope you can shed some light on this problem. I really like this plugin (though there is one more issue that's probably my fault, I'll look into it myself before bothering you) and I'm glad its developer is still active.

**Comment by Birch-san** [comment]
Heyo, thanks for trying out juicysfplugin.

Regarding this theory:

> something was either added or taken out of libfluidsynth-2.dll recently

I don't make [fluidsynth](http://www.fluidsynth.org/), and haven't changed anything about how I distribute it for 16 months.

But I think the issue is not specifically a problem with fluidsynth or juicysfplugin. I think it's likely to be an installation issue.

[`g_file_test`](https://developer.gnome.org/glib/stable/glib-File-Utilities.html#g-file-test) is a symbol in the [glib](https://developer.gnome.org/glib/) library which fluidsynth links against.

I think the error message means:  
_`libfluidsynth-2.dll` cannot find `libglib-2.0-0.dll`._

First let's try and take Ardour out of the equation, and do a simpler test. Does juicysfplugin.exe work? It's usually located at `"C:\Program Files\Birchlabs\juicysfplugin\juicysfplugin.exe"`. Open it, and try to load a soundfont file.

If juicysfplugin.exe encounters the same error, then that further corroborates the theory "`libglib-2.0-0.dll` is not in expected location".

Do you see `libglib-2.0-0.dll` located at `C:\Windows\System32\libglib-2.0-0.dll`, with filesize of about 1.11 MB? If not, then that explains the problem.

Otherwise: have you tried deleting the VST (`"C:\Program Files\Common Files\VST\juicysfplugin.dll"`) and using the VST3 version instead? (I think I install this to `"C:\Program Files\Common Files\VST3\juicysfplugin.vst3"`, but check where Ardour is configured to look for VST3 plugins).

If there's anything missing from your installation, or if you'd like to double-check "which files should exist in which locations": refer to my comment here which explains how to install manually: https://github.com/Birch-san/juicysfplugin/issues/20#issuecomment-723877163

**Comment by DaforLynx** [comment]
I probably should have mentioned the Juicy SF works in every other program I've tried, those being Ableton and VSTi host. The standalone application also works fine. `libglib-2.0-0.dll`  is in the correct spot with the correct filesize, and `libfluidsynth-2.dll` is also there. I've tried reinstalling the program, but that doesn't work. So maybe it's an issue with how Ardour handles VSTs; in fact, the evidence points to the very thing. For all I know I could try again today and it might work (probably not).
